server/src/controllers/song.controllers.ts

```diff
@@ -0,0 +1,106 @@
+import { Request, RequestHandler, Response } from 'express';
+import { Song }  from '../models/song.model';
+import { SongDocument } from '../types/song.types';
+
+const sendResponse = (res: Response, statusCode: number, payload: SongDocument[] | SongDocument | string ) => {
+  return res.status(statusCode).json(payload);
+};
+
+const createSong: RequestHandler = async (req: Request, res: Response): Promise<SongDocument | string> => {
+  const { ...toCreate } = req.body;
+
+  if (Object.keys(toCreate).length > 0) {
+    try {
+      const data = await Song.create(toCreate);
+
+      if (data) {
+        sendResponse(res, 200, data);
+      } else {
+        sendResponse(res, 404, 'Song not created');
+      }
+    } catch (error: any) {
+      sendResponse(res, 500, error.message);
+    }
+  } else {
+    sendResponse(res, 400, 'Missing required fields' );
+  }
+  return '';
+};
+
+const getSong: RequestHandler = async (req: Request, res: Response): Promise<SongDocument[] | SongDocument | string> => {
+    const { id: songId } = req.params;
+
+    if (songId) {
+      try {
+        const data = await Song.findOne({ _id: songId }).exec();
```

Comment:
woud be better if u can define the data type here, along with the other data types in the other APIs.
As of now, this is implicitly any type.